Raven.Database/Bundles/MoreLikeThis/MoreLikeThis.cs

```diff
@@ -802,7 +803,20 @@ internal PriorityQueue<object[]> RetrieveTerms(RavenJObject jsonDocument)
 
                 if (value == null) continue;
 
-                AddTermFrequencies(new StringReader(value.ToString()), words, fieldName);
+                switch (value.Type)
+                {
+                    case JTokenType.Array:
+                        foreach (var item in (RavenJArray)value)
+                        {
+                            if (item.Type != JTokenType.String) break;
+                            AddTermFrequencies(new StringReader(item.Value<string>()), words, fieldName);
+                        }
+                        break;
+                    case JTokenType.String:
+                        AddTermFrequencies(new StringReader(value.Value<string>()), words, fieldName);
+                        break;
+                    default: continue;
```

Comment:
Just for the case where the user ask to compare a given field and we are not able to do it. For example, the ```ComplexObj``` field will be ignored in this case:
```C#
session.Advanced.MoreLikeThis<IndexDocument, MapReduceIndex>(
    new MoreLikeThisQuery
    {
        Document = "{ "Tags": ["javascript"], "ComplexObj": { "Something": "some value" }  }",
        Fields = new[] { "Tags", "ComplexObj" }
    });
```